Source/Wpf/Prism.Wpf/Interactivity/InteractionRequest/InteractionRequest.cs

```diff
@@ -47,10 +48,59 @@ public void Raise(T context, Action<T> callback)
         /// if the InteractionRequest is unhandled.
         /// </summary>
         /// <param name="context">The context for the interaction request.</param>
-        /// <returns>The context after the request has been handled by the UI.</returns>
+        /// <returns>
+        /// The context after the request has been handled by the UI.
+        /// </returns>
         public Task<T> RaiseAsync(T context)
         {
-            return CallbackHelper.AwaitCallbackResult<T>(callback => this.Raise(context, callback));
+            return this.RaiseAsync(context, false);
+        }
+
+        /// <summary>
+        /// Fires the Raised event asynchronously. Please note that this request may never return
+        /// if the InteractionRequest is unhandled.
+        /// </summary>
+        /// <param name="context">The context for the interaction request.</param>
+        /// <param name="executeSynchronously">
+        /// Has no effect if the popup window is not modal.
+        /// If the popup window is modal and executeSynchronously is true, returns already completed task.
+        /// That is, when the method returns, the popup window has already been closed.
+        /// </param>
+        /// <returns>
+        /// The context after the request has been handled by the UI.
+        /// </returns>
+        public Task<T> RaiseAsync(T context, bool executeSynchronously)
+        {
+            if (executeSynchronously)
+            {
+                return CallbackHelper.AwaitCallbackResult<T>(callback => this.Raise(context, callback));
+            }
+            else
+            {
+                return this.RaiseImplAsync(context);
+            }
+        }
+
+        private async Task<T> RaiseImplAsync(T context)
+        {
+            // SynchronizationContext exists in actual program running on GUI Dispatcher thread.  Make sure
+            // it exists in tests.
+            if (SynchronizationContext.Current == null)
```

Comment:
We can remove that, but the developers will have a lot of troubles when the method will throw in their tests, they will figure out why, then they will setup the context...